src/lib/y2network/config_reader/sysconfig_interfaces.rb

```diff
@@ -0,0 +1,154 @@
+# Copyright (c) [2019] SUSE LLC
+#
+# All Rights Reserved.
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of version 2 of the GNU General Public License as published
+# by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+# more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, contact SUSE LLC.
+#
+# To contact SUSE LLC about this file by physical or electronic mail, you may
+# find current contact information at www.suse.com.
+
+require "yast"
+require "y2network/interface"
+require "y2network/virtual_interface"
+require "y2network/physical_interface"
+require "y2network/connection_config/ethernet"
+
+Yast.import "LanItems"
+Yast.import "NetworkInterfaces"
+
+module Y2Network
+  module ConfigReader
+    # This class reads interfaces configuration from sysconfig files
+    #
+    # * Physical interfaces are read from the hardware.
+    # * Virtual interfaces + Connections are read from sysconfig.
+    #
+    # @see Y2Network::Interface
+    # @see Y2Network::Connection::Connection
+    class SysconfigInterfaces
+      # Returns the interfaces and connections configuration
+      #
+      # @todo Should we use different readers?
+      # @todo Are virtual interfaces coming from connections only?
```

Comment:
well, if it is existing one, then it also exists on system e.g. hwinfo can return it, but not much info about it.